### PR TITLE
Add support for conditional rendering to jsx-multiline-wrap

### PR DESCRIPTION
**Pull request by jseminck**
Main issue: https://github.com/yannickcr/eslint-plugin-react/issues/1282

I tried to make the fixer better so the output has the parens in the correct place. Probably, need to add a few more tests, but I think the current solution looks quite solid.

The rule is defaulted to `false` initially. If enabling it by default, a bigger version bump would be required?

See also some comments on another issue regarding `jsx-multiline-wrap`: https://github.com/yannickcr/eslint-plugin-react/issues/1207#issuecomment-313086986

### CODE REVIEW

**Comment by jseminck** [comment]
FYI broken build doesn't seem to be related to this branch. Currently when I try to build master I get the same error messages, even though the latest build on travis for master was successful (21hrs ago: https://travis-ci.org/yannickcr/eslint-plugin-react/builds/250125668)

File: docs/rules/jsx-wrap-multilines.md [hunk 1]
```diff
@@ -118,3 +120,27 @@ var hello = () => (
...
+<div>
+  {myExpression &&
+    <div>
+      <p>Hello World</p>
+    </div>
+  }
+</div>
+```
+
+The following patterns are not considered warnings when configured `{arrow: true}`.
```
Comment (deecewan) [inline docs/rules/jsx-wrap-multilines.md]: do you mean `{conditional: true}`?

**Comment by deecewan** [comment]
I know that it'd require a bigger version bump, but to me it'd make more sense to have all the rules enabled by default.

Also, while you're in there, feel like adding ternaries?

**Comment by ljharb** [comment]
Enabling rules by default should wait for a separate semver-major PR, released after the non-major changes are released.

**Comment by marcdavi-es** [comment]
*** sits wriggling with excitement waiting for the merge ***

**Comment by ljharb** [comment]
Sorry for the inconvenience; I've just merged a refactor to start using arrow functions. Would you mind rebasing?

**Comment by jseminck** [comment]
Are there some problems regarding for this PR to be moved forward? 

I'd like to take a further look at this rule because I currently joined a project where we use a lot of conditional rendering and `eslint-plugin-react` doesn't always fix things as I expect it to. 😄

**Comment by jseminck** [comment]
Seems like this was added in another PR, hence closing.

To achieve this functionality, use the option: `{logical: 'parens'}` or `{logical: 'parens-new-line'}`.

**Comment by ljharb** [comment]
@jseminck sorry about the conflict :-)

**Pull request closed without merging**